File: JCleanArchitectureApp/presentation/src/main/java/com/woowrale/jcleanarchitecture/di/components/MainComponent.java
package com.woowrale.jcleanarchitecture.di.components;

import com.woowrale.jcleanarchitecture.di.modules.MainModule;
import com.woowrale.jcleanarchitecture.ui.base.BaseActivity;
import com.woowrale.jcleanarchitecture.ui.search.home.SearchActivity;
import com.woowrale.jcleanarchitecture.ui.search.local.LocalSearchActivity;
import com.woowrale.jcleanarchitecture.ui.search.remote.RemoteSearchActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {MainModule.class})
public interface MainComponent {

    void inject(BaseActivity activity);

    void inject(SearchActivity activity);

    void inject(LocalSearchActivity activity);

    void inject(RemoteSearchActivity activity);

}

File: KCleanArchitectureApp/appdagger/src/main/java/com/woowrale/kcleanarchitecture/di/components/MainComponent.kt
package com.woowrale.kcleanarchitecture.di.components

import com.woowrale.kcleanarchitecture.di.modules.LocalModule
import com.woowrale.kcleanarchitecture.di.modules.MainModule
import com.woowrale.kcleanarchitecture.di.modules.RemoteModule
import com.woowrale.kcleanarchitecture.ui.base.BaseActivity
import com.woowrale.kcleanarchitecture.ui.details.DetailsActivity
import com.woowrale.kcleanarchitecture.ui.search.home.SearchActivity
import com.woowrale.kcleanarchitecture.ui.search.local.LocalSearchActivity
import com.woowrale.kcleanarchitecture.ui.search.remote.RemoteSearchActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [MainModule::class, RemoteModule::class, LocalModule::class])
interface MainComponent {

    fun inject(activity: BaseActivity)

    fun inject(activity: SearchActivity)

    fun inject(activity: LocalSearchActivity)

    fun inject(activity: RemoteSearchActivity)

    fun inject(activity: DetailsActivity)

}